### PR TITLE
Update integration tests to use vm_service_lib

### PR DESCRIPTION
**Pull request by DanTup**
This switches the integration tests to the `vm_service_lib` which is a lighter wrapper over the the VM service.

I originally used vm_service_client because it was in the dependencies list, but it seems to be a transitive dependency from `test` (which we flatten into all pubspecs). We've been blocked a few times (and [still are for the debugger tests](https://github.com/dart-lang/vm_service_client/issues/41)) because the extra functionality added by `vm_service_client` sometimes requires updates with changes to Dart and right now we're pinned to a version of `test` that makes it more complicated to get the updates through.

Most of the changes are a very simple mapping, though `waitForPause` got slightly more complicated since we have to handle waiting for the pause event and checking the isolates existing state ourselves.

### CODE REVIEW

**Comment by DanTup** [comment]
I guess adding `dev_dependencies` requires running `flutter update-packages --force-upgrade` too!

I think I'll need to wait until #23755 lands then, since running `update-packages` tries to update `test` and adds `test_api` which I think I probably don't want to do. I'll rebase this and run it after #23755 lands.

**Comment by DanTup** [comment]
@devoncarew this has been rebased and the bots are all green :-)

**Review by devoncarew — APPROVED** [review]
👍 

lgtm when the flutter build is green

**Pull request merged**
